Obtained from:	https://paste.opensuse.org/view/raw/bf6caa94

From 01f59c31a729b7253a914939d591663983235b25 Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Thu, 21 Jul 2022 09:02:00 +0200
Subject: [PATCH] 3rdparty: Fix build on unsupported platforms

PPC, PPC64 and x86 require additional build flags to use pffft and eigen.
---
 src/resonance-audio/CMakeLists.txt | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/resonance-audio/CMakeLists.txt b/src/resonance-audio/CMakeLists.txt
index 1d26cfd..1c36c87 100644
--- src/resonance-audio/CMakeLists.txt
+++ src/resonance-audio/CMakeLists.txt
@@ -206,6 +206,25 @@ qt_internal_add_3rdparty_library(BundledResonanceAudio
         ${SADIE_HRTFS_DIR}
         ../3rdparty/eigen
 )
+
+# Required by pffft on certain PowerPC archs
+qt_internal_extend_target(BundledResonanceAudio CONDITION GCC AND (${CMAKE_SYSTEM_PROCESSOR} MATCHES "(ppc|ppc64)$")
+    COMPILE_OPTIONS
+        -maltivec
+)
+
+# Required by eigen on certain PowerPC archs
+qt_internal_extend_target(BundledResonanceAudio CONDITION (${CMAKE_SYSTEM_PROCESSOR} MATCHES "(ppc|ppc64)$")
+    COMPILE_OPTIONS
+        -mvsx
+)
+
+# Use fallback mode where SSE is not available
+qt_internal_extend_target(BundledResonanceAudio CONDITION (${CMAKE_SYSTEM_PROCESSOR} MATCHES "i[3-6]86$")
+    COMPILE_OPTIONS
+        -DPFFFT_SIMD_DISABLE
+)
+
 qt_disable_warnings(BundledResonanceAudio)
 qt_set_symbol_visibility_hidden(BundledResonanceAudio)
 
-- 
2.37.0

